internal/core/core.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package core provides a validator for being able to
+// check for core files on DUT's before and after test
+// modules runs.
+package core
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/eventlis"
+	"google.golang.org/grpc"
+
+	fpb "github.com/openconfig/gnoi/file"
+	opb "github.com/openconfig/ondatra/proto"
+)
+
+var (
+	vendorCoreFilePath = map[opb.Device_Vendor]string{
+		opb.Device_JUNIPER: "/var/core/",
+		opb.Device_CISCO:   "/misc/disk1/",
+		opb.Device_NOKIA:   "/var/core/",
+		opb.Device_ARISTA:  "/var/core/",
+	}
+	vendorCoreFileNamePattern = map[opb.Device_Vendor]*regexp.Regexp{
+		opb.Device_JUNIPER: regexp.MustCompile(".*.tar.gz"),
+		opb.Device_CISCO:   regexp.MustCompile("/misc/disk1/.*core.*"),
+		opb.Device_NOKIA:   regexp.MustCompile("/var/core/coredump-.*"),
+		opb.Device_ARISTA:  regexp.MustCompile("/var/core/core.*"),
+	}
+)
+
+var (
+	Validator validator
+)
+
+type fileInfo struct {
+	Name     string
```

Comment:
These fields don't need to be capitalized/exported

Reply:
since they only make testing and templating harder i would prefer them to stay